hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1314,6 +1346,7 @@ func machineTemplateBuilders(hcluster *hyperv1.HostedCluster, nodePool *hyperv1.
 	}
 	template.SetNamespace(manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name).Name)
 	template.SetName(nodePool.GetName())
+	template.SetAnnotations(map[string]string{nodePoolAnnotation: client.ObjectKeyFromObject(nodePool).String()})
```

Comment:
this actually needs to be set for each mutateTemplate function for it to take effect.
Then can we add a check in TestMachineTemplateBuilders to validate this is set?

Reply:
I added a check in the UT as you suggested and it works without setting it in each mutateTemplate function.  Why is it necessary?

Reply:
I see, it works because it's used by createOrUpdate at creation time. However if after the machine template exists I go and manually edit it and drop the annotation, the controller wouldn't be able to set it back. That can actually be validated with another testcase for TestMachineTemplateBuilders that takes a preexisting machinetemplate without the annotation

Reply:
Once we have the testcase we can test driven move to  a `createOrUpdateWithAnnotationFactory` approach as we do in hostedCluster.

Reply:
You're right.  I added a test case and it failed until I moved to each mutate function.